### PR TITLE
Update codebase to PHP 7.4

### PR DESCRIPTION
**Pull request by TavoNiievez**
... also adds support for PHP 8.1

### CODE REVIEW

**Comment by vansari** [comment]
Thanks for your contribution 🥳🤘

Should we release this changes with a new major or a new minor version?!

In my opinion we should use a new major release.

**Comment by TavoNiievez** [comment]
Hi @vansari, and thanks ;-)

I touched the interfaces...

So it should definitely be a major release.

**Review by DavertMik — APPROVED** [review]

**Comment by TavoNiievez** [comment]
@vansari this is ready to merge

File: src/Splitter/SplitTestsByTimeTask.php [hunk 1]
```diff
@@ -57,7 +61,7 @@ public function run(): void
 
         arsort($testsWithTime);
 
-        for ($i = 0; $i < $this->numGroups; $i++) {
+        for ($i = 0; $i < $this->numGroups; ++$i) {
```
Comment (vansari) [inline src/Splitter/SplitTestsByTimeTask.php:64]: Why did you change it from post-increment to pre-increment?
This is not the only place :-)

Comment (TavoNiievez) [reply src/Splitter/SplitTestsByTimeTask.php:64]: Short answer: Performance. Pre-increment is faster than post-increment.
Long answer: See Explanation of why [here](https://stackoverflow.com/a/1756094) and [here](https://piotrzatorski.medium.com/pre-and-post-increment-decrement-how-do-they-work-7319b1ddef1d), or a performance test [here](https://syframework.alwaysdata.net/pre-increment).

Comment (DavertMik) [reply src/Splitter/SplitTestsByTimeTask.php:64]: I don't see that performance improvement does matter here... 
But it is not a big deal. I think this PR can be merged

**Comment by vansari** [comment]
After the Merge from @DavertMik, can you please resolve the conflicts @TavoNiievez ? 😃
